Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h

```diff
@@ -190,6 +190,11 @@ typedef NS_ENUM(NSInteger, BTPayPalRequestUserAction) {
  */
 @property (nonatomic, nullable, strong) UIWindow *activeWindow;
 
+/**
+ Optional: If `true`, the customer will be prompted to consent to a billing agreement during PayPal Checkout flows.
+ */
+@property (nonatomic) BOOL requestBillingAgreement;
```

Comment:
It looks like there are a [few](https://github.com/braintree/braintree_ios/blob/0c3860281da41f3b95eca3556ae65d3f0f7eb901/Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h#L142) [other](https://github.com/braintree/braintree_ios/blob/0c3860281da41f3b95eca3556ae65d3f0f7eb901/Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h#L147) [properties](https://github.com/braintree/braintree_ios/blob/0c3860281da41f3b95eca3556ae65d3f0f7eb901/Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h#L174) that only apply to one of the two PayPal flows. I don't know if we'd want to do this refactor now, but I wonder if the ideal state would be to have a `BTPayPalRequest` base class, and two subclasses: `BTPayPalCheckoutRequest` and `BTPayPalBillingAgreementRequest`. That way, it would be a bit more clear which properties apply to only one flow and which apply to both.

Reply:
🤔 I like that